tespy/components/customs.py

```diff
@@ -0,0 +1,818 @@
+# -*- coding: utf-8
+
+"""Module for custom components.
+
+Components in this module:
+
+    - :func:`tespy.components.customs.orc_evaporator`
+
+This file is part of project TESPy (github.com/oemof/tespy). It's copyrighted
+by the contributors recorded in the version control history of the file,
+available from its original location tespy/components/customs.py
+
+SPDX-License-Identifier: MIT
+"""
+
+import logging
+
+import numpy as np
+
+from tespy.components.components import component
+
+from tespy.tools.data_containers import dc_cc, dc_cp, dc_simple, dc_gcp
+from tespy.tools.fluid_properties import (
+        h_mix_pT, s_mix_ph, v_mix_ph, visc_mix_ph, T_mix_ph,
+        dh_mix_dpQ, h_mix_pQ, T_bp_p, memorise
+        )
+from tespy.tools.helpers import lamb, single_fluid
+
+# %%
+
+
+class orc_evaporator(component):
+    r"""
+    Class orc_evaporator is the evaporator component in
+    the Organic Rankine Cycle (ORC). Generally, the hot side
+    of the geo-fluid from the geothermal wells keeps 2-phase.
+    In order to fully use the energy in the geo-fluid,
+    there are 2 inlets at the hot side.
+
+    The ORC evaporator represents counter current evaporators. Both, 2 hot
+    and 1 cold side of the evaporator, are simulated.
+
+    Equations
+
+        **mandatory equations**
+
+        - :func:`tespy.components.customs.orc_evaporator.fluid_func`
+        - :func:`tespy.components.customs.orc_evaporator.mass_flow_func`
+
+        - :func:`tespy.components.customs.orc_evaporator.energy_func`
+
+        .. math::
+
+            0 = p_{1,in} \cdot pr1 - p_{1,out}\\
+            0 = p_{2,in} \cdot pr2 - p_{2,out}\\
+            0 = p_{3,in} \cdot pr3 - p_{3,out}
+
+        - :func:`tespy.components.customs.orc_evaporator.zeta_func`
+        - :func:`tespy.components.customs.orc_evaporator.zeta2_func`
+        - :func:`tespy.components.customs.orc_evaporator.zeta3_func`
+
+        **additional equations**
+
+        - :func:`tespy.components.customs.orc_evaporator.additional_equations`
+
```

Comment:
We can skip the usage of the additional equations. The additional equations are intended for components, that inherit some equations from a parent component but bring individual properties. For example: The fluid's state at the hot side of a heat_exchanger is not predefined. In a condenser however, the fluid is in saturated liquid state at that point. Therefore the condenser brings one additional equation to the base functionality of the heat_exchanger.

Reply:
Thanks for the explanation. Deleted.